### ISSUE TITLE
Use search Clipboard

### ISSUE THREAD

**Issue by dwt** [zed-industries/zed#7726]
### Check for existing issues

- [X] Completed

### Describe the bug / provide steps to reproduce it

Zed doesn't seem to use the shared search clipboard. This means I cannot for example select some text in the editor, the browser or the terminal and hit command-e to copy that text into the search clipboard that all applications share, switch over to zed, and hit either `command-f` to search for it in the current file, or `command-shift-f` to search for it in all files, like I would with other editors.

### Environment

Zed: v0.121.7 (Zed)
OS: macOS 14.3.1
Memory: 32 GiB
Architecture: aarch64

### If applicable, attach your `~/Library/Logs/Zed/Zed.log` file to this issue.
If you only need the most recent lines, you can run the `zed: open log` command palette action to see the last 1000.

_No response_

**Comment by jansol** [comment]
TIL this is actually a thing. In Firefox on Linux, if I select some text in the browser and hit ctrl+f, the selected text gets pasted in the search field. Alas this does not seem to work with the primary selection or clipboard from other apps.

Seems the search clipboard simply replaces this? Also it seems that Discord doesn't support the shared search clipboard either, bummer (it pops up the emoji selector on cmd+e instead).

**Comment by dwt** [comment]
Yes, this is still reproducible, and quite sorely missed, as the search keyboard is an integral part of my workflow.

**Comment by cole-miller** [comment]
Thanks for the idea---this seems like something we should talk through in the discussions.
